### ISSUE TITLE
make littleflash a submodule

### ISSUE THREAD

**Issue by phatpaul** [lllucius/esp32_littleflash#1]
Would you please split this up so I can include just the littleflash part in my ESP-IDF project as a git submodule?

I.e. just like you did with extflash.

I want to include it in my PR https://github.com/chmorgan/esphttpd-freertos/pull/12 but the maintainer doesn't want any duplicated code, just submodules.

**Comment by lllucius** [comment]
Should be good now.  Let me know if it doesn't work as you wanted.

**Comment by phatpaul** [comment]
Thanks, but I think there some recursion hell going on.

```
In file included from C:/msys32/home/labview/esphttpd-freertos/main/user_main.c:76:0:
C:/msys32/home/labview/esphttpd-freertos/components/littleflash/components/littleflash/include/littleflash.h:24:22: fatal error: extflash.h: No such file or directory
compilation terminated.
```

that file is at 
"C:\msys32\home\labview\esphttpd-freertos\components\littleflash\components\extflash\components\extflash\include\extflash.h"

Maybe is there not allowed to be multiple components with the same name?  (there are 2 levels of extflash)

**Comment by lllucius** [comment]
Hmmm, I wonder why it failed there.  I've attached a directory listing showing the tree I tested it in and it built fine.  I suspect it's because I have an "extflash" in the httpd/components directory.  I don't believe the littlefs/components/extflash fella is used when you're using this sort of hieirarchy.

[tree.txt](https://github.com/lllucius/esp32_littleflash/files/2968157/tree.txt)

**Comment by lllucius** [comment]
Did you happen to do the "git submodule init/update" in the littleflash directory?  I set up a clean tree and didn't do the "init/update" in the littleflash directory and littleflash/components/extflash was not pulled down, so it wouldn't pollute your tree with too much extra stuff.

**Comment by phatpaul** [comment]
Oh, I did not add extflash to my first-level components directory (since I wasn't planning to use it).
I tried adding it to see if it compiles, but it gives new errors. 
First one is below.  Seems it is not expecting C++?
```
C:/msys32/home/labview/esphttpd-freertos/components/extflash/components/extflash/include/extflash.h:52:1: error: unknown type name 'class'
 class ExtFlash
 ^
```

**Comment by phatpaul** [comment]
Sorry I didn't realize your port was in C++.
I guess there is not an easy way to use it from a C project? (without converting the project main.c to cpp)

**Comment by lllucius** [comment]
Yes, that would be about the only way.  But, it should work just fine.  You might have a few minor issues to address.

Or you could break out the littlefs code you have in “main” into a separate .cpp file and then have your main code call that.
